### PR TITLE
[C#] Idiomatic formatting

### PR DESCRIPTION
**Pull request by yowl**
Generic types begin with `T`
Methods start with uppercase
some block `{` moved to own line.

draft as behind #1070

### CODE REVIEW

File: crates/csharp/src/lib.rs [hunk 1]
```diff
@@ -3292,28 +3333,111 @@ fn is_primitive(ty: &Type) -> bool {
...
 
 trait ToCSharpIdent: ToOwned {
+    fn csharp_keywords() -> Vec<&'static str>;
     fn to_csharp_ident(&self) -> Self::Owned;
+    fn to_csharp_ident_upper(&self) -> Self::Owned;
 }
 
 impl ToCSharpIdent for str {
+    // Source: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/
+    // TODO: Repace with actual keywords
```
Comment (silesmo) [inline crates/csharp/src/lib.rs:3343]: This todo should be have been resolved a long time ago I think?

**Comment by silesmo** [comment]
Marking this as ready for review as #1070 has now been merged.

**Review by silesmo — APPROVED** [review]
Thank you! The only thing potentially is if the TODO I commented on should be removed. So leaving it up to you @yowl If that is the case or not.

File: crates/csharp/src/lib.rs (hunk 1 above)
Comment (yowl) [reply crates/csharp/src/lib.rs]: Correct, I have now removed this comment.

**Comment by yowl** [comment]
Thanks for the review

**Pull request merged**